src/test/scala/advancepassengerinfo/importer/persistence/MockPersistence.scala

```diff
@@ -32,3 +34,44 @@ case class MockPersistence(probe: ActorRef) extends Persistence {
 
   override def jsonHasBeenProcessed(zipFileName: String, jsonFileName: String): Future[Boolean] = Future.successful(false)
 }
+
+case class MockPersistenceWithRecovery(probe: ActorRef) extends Persistence {
+  private var isFirstCall: Boolean = true
+
+  override def persistManifest(jsonFileName: String, manifest: VoyageManifest): Future[Option[Int]] =
+    if (isFirstCall) {
+      isFirstCall = false
+      Future.failed(new Exception("db persistManifest exception"))
+    } else {
+      probe ! ManifestCall(jsonFileName, manifest)
+      Future.successful(Option(1))
+    }
+
+  override def persistJsonFile(zipFileName: String, jsonFileName: String, successful: Boolean, dateIsSuspicious: Boolean): Future[Int] =
+    if (isFirstCall) {
+      isFirstCall = false
+      Future.failed(new Exception("db persistJsonFile exception"))
+    } else {
+      probe ! JsonFileCall(zipFileName, jsonFileName, successful, dateIsSuspicious)
+      Future.successful(1)
+    }
+
+  override def persistZipFile(zipFileName: String, successful: Boolean): Future[Boolean] =
+    if (isFirstCall) {
+      isFirstCall = false
+      Future.failed(new Exception("db persistZipFile exception"))
+    } else {
+      probe ! ZipFileCall(zipFileName, successful)
+      Future.successful(true)
+    }
+
+  override def lastPersistedFileName: Future[Option[String]] =
+    Future.successful(Option("1.zip"))
+
+  override def jsonHasBeenProcessed(zipFileName: String, jsonFileName: String): Future[Boolean] =
+    if (isFirstCall) {
+      isFirstCall = false
+      Future.failed(new Exception(s"db jsonHasBeenProcessed exception $isFirstCall"))
```

Comment:
`$isFirstCall` will always be false here as it's set on the line above

Reply:
I mean literally on the line above it says `isFirstCall = false` so where you're outputting it in the exception it can never be anything but false